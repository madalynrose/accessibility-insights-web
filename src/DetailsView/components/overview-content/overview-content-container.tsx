// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { AssessmentsProvider } from 'assessments/types/assessments-provider';
import { NamedFC } from 'common/react/named-fc';
import { HyperlinkDefinition } from 'common/types/hyperlink-definition';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { DetailsViewPivotType } from 'common/types/store-data/details-view-pivot-type';
import { FeatureFlagStoreData } from 'common/types/store-data/feature-flag-store-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import * as React from 'react';

import { OverviewSummaryReportModel } from 'reports/assessment-report-model';
import { AssessmentReportSummary } from 'reports/components/assessment-report-summary';
import { GetAssessmentSummaryModelFromProviderAndStoreData } from 'reports/get-assessment-summary-model';
import { GetQuickAssessSummaryModelFromProviderAndStoreData } from 'reports/get-quick-assess-summary-model';

import { TargetChangeDialog, TargetChangeDialogDeps } from '../target-change-dialog';
import styles from './overview-content-container.scss';
import { OverviewHeading } from './overview-heading';
import { OverviewHelpSection, OverviewHelpSectionDeps } from './overview-help-section';

const linkDataSource: HyperlinkDefinition[] = [
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2082219',
        text: 'Getting started',
    },
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2082220',
        text: 'How to complete a test',
    },
    {
        href: 'https://go.microsoft.com/fwlink/?linkid=2077941',
        text: 'Ask a question',
    },
    {
        href: 'https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/',
        text: 'New WCAG 2.1 success criteria',
    },
];

export type OverviewContainerDeps = {
    assessmentsProvider: AssessmentsProvider;
    getAssessmentSummaryModelFromProviderAndStoreData: GetAssessmentSummaryModelFromProviderAndStoreData;
    getQuickAssessSummaryModelFromProviderAndStoreData: GetQuickAssessSummaryModelFromProviderAndStoreData;
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    assessmentsProviderWithFeaturesEnabled: (
        assessmentProvider: AssessmentsProvider,
        flags: FeatureFlagStoreData,
    ) => AssessmentsProvider;
    assessmentsProviderForRequirements: (
        assessmentsProvider: AssessmentsProvider,
        requirementKeys: string[],
    ) => AssessmentsProvider;
    quickAssessRequirementKeys: string[];
} & OverviewHelpSectionDeps &
    TargetChangeDialogDeps;

export interface OverviewContainerProps {
    deps: OverviewContainerDeps;
    assessmentStoreData: AssessmentStoreData;
    tabStoreData: TabStoreData;
    featureFlagStoreData: FeatureFlagStoreData;
    selectedPivot: DetailsViewPivotType;
}

export const overviewContainerAutomationId = 'overviewContainerAutomationId';

export const OverviewContainer = NamedFC<OverviewContainerProps>('OverviewContainer', props => {
    const { deps, assessmentStoreData, tabStoreData, featureFlagStoreData, selectedPivot } = props;
    const {
        assessmentsProvider,
        getAssessmentSummaryModelFromProviderAndStoreData,
        getQuickAssessSummaryModelFromProviderAndStoreData,
        assessmentsProviderWithFeaturesEnabled,
        assessmentsProviderForRequirements,
        quickAssessRequirementKeys,
    } = deps;
    const prevTarget = assessmentStoreData.persistedTabInfo;
    const currentTarget = {
        id: tabStoreData.id,
        url: tabStoreData.url,
        title: tabStoreData.title,
    };
    let filteredProvider = assessmentsProviderWithFeaturesEnabled(
        assessmentsProvider,
        featureFlagStoreData,
    );
    let summaryData: OverviewSummaryReportModel;

    if (selectedPivot === DetailsViewPivotType.assessment) {
        summaryData = getAssessmentSummaryModelFromProviderAndStoreData(
            filteredProvider,
            assessmentStoreData,
        );
    } else {
        filteredProvider = assessmentsProviderForRequirements(
            filteredProvider,
            quickAssessRequirementKeys,
        );
        summaryData = getQuickAssessSummaryModelFromProviderAndStoreData(
            filteredProvider,
            assessmentStoreData,
            quickAssessRequirementKeys,
        );
    }

    return (
        <div data-automation-id={overviewContainerAutomationId} className={styles.overview}>
            <TargetChangeDialog deps={deps} prevTab={prevTarget} newTab={currentTarget} />
            <section className={styles.overviewTextSummarySection}>
                <OverviewHeading />
                <AssessmentReportSummary summary={summaryData} selectedPivot={selectedPivot} />
            </section>
            <section className={styles.overviewHelpSection}>
                <OverviewHelpSection linkDataSource={linkDataSource} deps={deps} />
            </section>
        </div>
    );
});
