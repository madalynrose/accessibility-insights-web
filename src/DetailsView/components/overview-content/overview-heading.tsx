// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { OverviewHeadingIntroFactory } from 'DetailsView/components/overview-content/overview-heading-intro';
import * as React from 'react';

import { NamedFC } from '../../../common/react/named-fc';
import styles from './overview-heading.scss';

export const overviewHeadingAutomationId = 'overview-heading';

export type OverviewHeadingProps = {
    getIntroComponent: OverviewHeadingIntroFactory;
};

export const OverviewHeading = NamedFC('OverviewHeading', (props: OverviewHeadingProps) => {
    return (
        <>
            <div
                className={styles.overviewHeading}
                data-automation-id={overviewHeadingAutomationId}
            >
                <h1>Overview</h1>
                {props.getIntroComponent()}
            </div>
        </>
    );
});
