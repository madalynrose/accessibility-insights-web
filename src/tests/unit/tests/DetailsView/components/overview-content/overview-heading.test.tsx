// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';

import { OverviewHeading } from '../../../../../../DetailsView/components/overview-content/overview-heading';

describe('OverviewHeading', () => {
    test('match snapshot', () => {
        const getIntroComponentStub = () => <div>INTRO COMPONENT</div>;
        const wrapper = shallow(<OverviewHeading getIntroComponent={getIntroComponentStub} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
