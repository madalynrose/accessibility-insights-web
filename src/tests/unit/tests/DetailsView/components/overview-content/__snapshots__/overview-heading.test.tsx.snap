// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewHeading match snapshot 1`] = `
<React.Fragment>
  <div
    className="overviewHeading"
    data-automation-id="overview-heading"
  >
    <h1>
      Overview
    </h1>
    <div>
      INTRO COMPONENT
    </div>
  </div>
</React.Fragment>
`;
